src/ducks/sep6WithdrawAsset.ts

```diff
@@ -0,0 +1,328 @@
+        const signedChallengeTransaction = sep10AuthSign({
+          secretKey,
+          networkPassphrase: networkConfig.network,
+          challengeTransaction,
+        });
+
+        // SEP-10 send
+        const token = await sep10AuthSend({
+          authEndpoint: webAuthTomlResponse.WEB_AUTH_ENDPOINT,
+          signedChallengeTransaction,
+        });
+
+        // Get SEP-12 fields
+        log.instruction({
+          title: "Making GET `/customer` request for user",
+        });
+
+        const sep12Fields = await collectSep12Fields({
+          publicKey,
+          token,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+        });
+
+        payload = {
+          ...payload,
+          kycServer: webAuthTomlResponse.KYC_SERVER,
+          token,
+        };
+
+        if (sep12Fields) {
+          payload = {
+            ...payload,
+            fields: { ...payload.fields, ...sep12Fields },
+          };
+        }
+      }
+
+      return payload;
+    } catch (error) {
+      const errorMessage = getErrorMessage(error);
+
+      log.error({
+        title: "SEP-6 withdrawal failed",
+        body: errorMessage,
+      });
+
+      return rejectWithValue({
+        errorString: errorMessage,
+      });
+    }
+  },
+);
+
+export const submitSep6DWithdrawFields = createAsyncThunk<
+  { status: ActionStatus; type: string; withdrawFields: AnyObject },
+  {
+    withdrawType: AnyObject;
+    infoFields: AnyObject;
+    customerFields: AnyObject;
+  },
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep6WithdrawAsset/submitSep6WithdrawFields",
+  async (
+    { withdrawType, infoFields, customerFields },
+    { rejectWithValue, getState },
+  ) => {
+    try {
+      const { secretKey } = accountSelector(getState());
+      const { data } = sepWithdrawSelector(getState());
+      const { kycServer, token } = data;
+
+      if (Object.keys(customerFields).length) {
+        await putSep12FieldsRequest({
+          fields: customerFields,
+          kycServer,
+          secretKey,
+          token,
+        });
+      }
+      return {
+        status: ActionStatus.CAN_PROCEED,
+        type: withdrawType.type,
+        withdrawFields: infoFields,
+      };
+    } catch (e) {
+      const errorMessage = getErrorMessage(e);
+
+      log.error({
+        title: errorMessage,
+      });
+
+      return rejectWithValue({
+        errorString: errorMessage,
+      });
+    }
+  },
+);
+
+export const sep6WithdrawAction = createAsyncThunk<
+  { withdrawResponse: Sep6WithdrawResponse; status: ActionStatus },
+  undefined,
+  { rejectValue: RejectMessage; state: RootState }
+>(
+  "sep6WithdrawAsset/sep6WithdrawAction",
+  async (_, { rejectWithValue, getState }) => {
+    try {
+      const { data } = accountSelector(getState());
+      const { claimableBalanceSupported } = settingsSelector(getState());
+      const publicKey = data?.id || "";
+      const { data: sep6data } = sepWithdrawSelector(getState());
```

Comment:
nit: missed a camelCase:
```suggestion
      const { data: sep6Data } = sepWithdrawSelector(getState());
```